### PR TITLE
Add memory request to provisioning pod.

### PR DESCRIPTION
**Pull request by twiest**
- [x] Add memory request to provisioning pod.
- [x] Add unit test
- [x] Manually test

### CODE REVIEW

**Comment by twiest** [comment]
/hold

Holding until manual testing is complete

File: pkg/install/generate.go [hunk 1]
```diff
@@ -384,6 +385,13 @@ func InstallerPodSpec(
 			Command:         []string{"/bin/sh", "-c"},
 			Args:            []string{hiveArg},
 			VolumeMounts:    volumeMounts,
+			Resources: corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
+					corev1.ResourceRequestsMemory: resource.Quantity{
+						Format: "838860800",
```
Comment (joelddiaz) [inline pkg/install/generate.go:391]: can you put a comment to explain what this magic-looking number is?

**Comment by twiest** [comment]
@joelddiaz I believe I've addressed your feedback. PTAL.

/hold cancel

**Pull request merged**